### PR TITLE
ssh2: fix for large file sizes

### LINKED ISSUE #66: large size files bug
gftp-2 0 19

gftp 2.0.19-5
in ssh2 connection to my website

I notice that the value of large files larger than 20 GB is wrong 

I have a big planet.bin file (30 GB) on my site
this is one openstreetmap navit map
36213412764 bytes

When i make ssh2 connection with gftp, the file size is on remote site
1853674396 bytes (that's false)

When i make ssh2 connection with filezilla, the file size is on remote site
36213412764 bytes (that's right)

you can see the picture

### PR DESCRIPTION
**Pull request by wdlkmpx**
see issue #66 for screenshots and discussion

closes #66

### CODE REVIEW

**Comment by wdlkmpx** [comment]
Int64_t is defined in stdint.h (c99):
https://en.cppreference.com/w/c/types/integer
http://www.cplusplus.com/reference/cstdint/

Gftp.h includes inttypes.h
https://stackoverflow.com/questions/7597025/difference-between-stdint-h-and-inttypes-h

Off_t: if the programming environment enables large files off_t is defined to be a signed, 64-bit integer.
http://www.ahay.org/RSF/book/rsf/manual/manual_html/node44.html
https://github.com/masneyb/gftp/blob/master/lib/gftp.h#L33

I can revert the commit that changes off_t to int64_t

File: lib/sshv2.c [hunk 1]
```diff
@@ -988,7 +988,9 @@ sshv2_buffer_get_int64 (gftp_request * request, sshv2_message * message,
   if ((ret = sshv2_buffer_get_int32 (request, message, 0, 0, &lonum)) < 0)
     return (ret);
 
-  snum = (gint64) lonum | ((gint64) hinum >> 32);
+  snum = (guint64) lonum | ((guint64) hinum << 32);
```
Comment (masneyb) [inline lib/sshv2.c:991]: How about casting both of these to off_t instead of guint64? Good catch with the bit shift error.

File: lib/sshv2.c [hunk 2]
```diff
@@ -988,7 +988,9 @@ sshv2_buffer_get_int64 (gftp_request * request, sshv2_message * message,
...
+
+  //fprintf(stderr, "%o %o\n%jd\n\n", lonum, hinum, snum); //debug
```
Comment (masneyb) [inline lib/sshv2.c:993]: Let's remove this debug line

File: src/gtk/listbox.c [hunk 3]
```diff
@@ -133,7 +133,7 @@ listbox_add_columns (gftp_window_data *wdata)
    /* icon */
    //renderer = gtk_cell_renderer_pixbuf_new ();
    renderer = g_object_new (GTK_TYPE_CELL_RENDERER_PIXBUF,
-                           "xalign", 0.5,     /* justify left */
+                           "xalign", 0.5,     /* justify center */
```
Comment (masneyb) [inline src/gtk/listbox.c:136]: I almost hate to ask this but maybe this should be in a separate PR. If it's too much of a pain, then I'll take this as is in this PR once those two other minor changes above are addressed. Thanks for your contributions!

**Comment by wdlkmpx** [comment]
sshv2_buffer_get_int64 is called only to set a new value to fle->size (off_t),

I changed off_t to gint64, both are compatible, otherwise there's a gcc warning:
guint64 = long long unsigned int
off_t = long long int

I think the 'size' should be guint64 or uint64_t, but some operations throw some gcc warnings, because they expect a signed 64 bit variable or something.

**Pull request merged**